src/treemap/split.js

```diff
@@ -0,0 +1,105 @@
+export default function(parent, x0, y0, x1, y1) {
+
+  parent.x0 = x0;
+  parent.x1 = x1;
+  parent.y0 = y0;
+  parent.y1 = y1;
+
+  // exit if no children
+  if (!(parent.children) || parent.children.length === 0) {
+    return;
+  }
+
+  splitTree(parent.children, parent.value, x0, y0, x1, y1);
```

Comment:
ok, I wondered about this assumption.  It sounds like my assumption could easily be violated in which case speed improvements will be dramatically reduced :(  I'll revert back to the original `sum` handling.